synapse/federation/sender/__init__.py

```diff
@@ -560,3 +577,35 @@ async def get_replication_rows(
         # Dummy implementation for case where federation sender isn't offloaded
         # to a worker.
         return [], 0, False
+
+    async def _wake_destinations_needing_catchup(self):
+        """
+        Wakes up destinations that need catch-up and are not currently being
+        backed off from.
+        Does so in a slow way (one every 5 seconds) to reduce load spikes.
+        """
+
+        last_processed = None  # type: Optional[str]
+
+        while True:
+            destinations_to_wake = await self.store.get_catch_up_outstanding_destinations(
+                last_processed
+            )
+
+            if not destinations_to_wake:
+                # finished waking all destinations!
+                break
```

Comment:
It could be nice to clear out the `_catchup_after_startup_timer` here (set it to `None`?)